### PR TITLE
Freud 2.10.0 release with `macos-arm64` builds.

### PR DESCRIPTION
**Pull request by joaander**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This is another attempt to see if the scikit-build cross-compilation for macos-arm64 works any better.

### CODE REVIEW

**Comment by joaander** [comment]
This requires changes to freud's CMake code. Will update and merge this after the next freud release.

**Comment by joaander** [comment]
ppc64le builds are failing without any log.
@conda-forge-admin, please restart ci

File: recipe/build_osx_arm64.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+echo "CONDA_PREFIX=${CONDA_PREFIX}"
+echo "CFLAGS=${CFLAGS}"
+echo "CXXFLAGS=${CXXFLAGS}"
+
+export CMAKE_PREFIX_PATH=${PREFIX}
+
+# Filter CMAKE_PREFIX_PATH out of CMAKE_ARGS because scikit-build needs to set it
+export SKBUILD_CONFIGURE_OPTIONS=$(echo $CMAKE_ARGS | sed -e 's/\-DCMAKE_INSTALL_PREFIX\=[^ ]* //g')
+
+# work around a bug in conda-forge where the installed cython gets an invalid #! line: "#!$BUILD_PREFIX/bin/python"
```
[the remark this answers is not in the document]
Comment (joaander) [reply recipe/build_osx_arm64.sh:19]: I find it very challenging to debug issues with prefixes in the conda-forge build as there appears to be some sort of output processor that converts `/path/to/conda-env/placeholder/placeholder/...` to $PREFIX *in the display*. For example, `echo $PREFIX` always outputs `$PREFIX`. Thus, in tests that `cat cython` for example, I can't tell whether there is really a literal `$PREFIX` in the file, or the actual absolute path. 

In the case of cython, the behavior I get without the patch is consistent with a literal `$PREFIX` existing in the file. I do not know what stage of the conda build and/or install process puts this here, so I wouldn't know where to submit an upstream bug.

**Comment by joaander** [comment]
For reference, the pip bug is discussed in https://github.com/pypa/pip/issues/11116 and https://github.com/conda-forge/mdanalysis-feedstock/pull/42. Once pip releases a fix we can update the version specification accordingly.

Vyas, I had to remove the tests because they are not included in the source tarball provided on pypi and there is no alternate source tarball. When such tests are added, we can uncomment the pytest line again. But for the moment we will be publishing untested binaries.

File: recipe/build_osx_arm64.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (joaander) [reply recipe/build_osx_arm64.sh:19]: Specifically, I don't know whether this is caused in upstream cython, in the cython conda-forge feedstock, in conda's installation relocation mechanism, in the conda-forge docker build scripts, or in some combination of the above.

**Review by tommy-waltmann — APPROVED** [review]
Looks good!

File: recipe/build_osx_arm64.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (vyasr) [reply recipe/build_osx_arm64.sh:19]: Yeah this process is difficult to debug. I don't think the problem can be in upstream Cython, and I think the relocation is the most likely culprit, but I don't know how to verify that it's not any other part of the conda process without spending a lot more time searching. The feedstock _looks_ correct to me, and when I download builds the directory is set correctly in the console entry point script, but that just means that there are lots of situations where they aren't currently running into this problem. If we get a chance later we can try to follow up, but I'm fine with getting this merged to unblock freud wheels.

File: recipe/build_osx_arm64.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+echo "PREFIX=${PREFIX}"
+echo "BUILD_PREFIX=${BUILD_PREFIX}"
+echo "CONDA_PREFIX=${CONDA_PREFIX}"
+echo "CFLAGS=${CFLAGS}"
+echo "CXXFLAGS=${CXXFLAGS}"
+
+export CMAKE_PREFIX_PATH=${PREFIX}
+
+# Filter CMAKE_PREFIX_PATH out of CMAKE_ARGS because scikit-build needs to set it
+export SKBUILD_CONFIGURE_OPTIONS=$(echo $CMAKE_ARGS | sed -e 's/\-DCMAKE_INSTALL_PREFIX\=[^ ]* //g')
```
Comment (vyasr) [inline recipe/build_osx_arm64.sh:17]: Why not just modify the `CMAKE_ARGS` directly in this script? There's nothing being built that uses them other than freud anyway, right?

**Review by b-butler — APPROVED** [review]
This looks good to me, though this is outside my general software knowledge.

File: recipe/build_osx_arm64.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export VERBOSE=1
```
Comment (vyasr) [inline recipe/build_osx_arm64.sh:1]: Do we want compilation to always be verbose?

**Review by vyasr — APPROVED** [review]
I'm good with moving forward here. We can do cleanup and add tests later if we're able to resolve the underlying Cython issues and get tests into our tarballs.

File: recipe/build_osx_arm64.sh (hunk 3 above)
Comment (joaander) [reply recipe/build_osx_arm64.sh:1]: It adds a moderate amount to the log size, but provides useful debugging information if the builds ever break in the future. That makes one less step to debugging a broken build.

**Pull request merged**